module/FinnaApi/src/FinnaApi/Controller/BazaarApiController.php

```diff
@@ -0,0 +1,198 @@
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Aida Luuppala <aida.luuppala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+namespace FinnaApi\Controller;
+
+use VuFindApi\Controller\ApiController;
+use VuFindApi\Controller\ApiInterface;
+use VuFindApi\Controller\ApiTrait;
+
+/**
+ * Bazaar API Controller
+ *
+ * Controls the Bazaar API functionality
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Aida Luuppala <aida.luuppala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+class BazaarApiController extends ApiController implements ApiInterface
+{
+    use ApiTrait;
+
+    /**
+     * Execute the request
+     *
+     * @param \Laminas\Mvc\MvcEvent $e Event
+     *
+     * @return mixed
+     * @throws Exception\DomainException
+     */
+    public function onDispatch(\Laminas\Mvc\MvcEvent $e)
+    {
+        // Add CORS headers and handle OPTIONS requests. This is a simplistic
+        // approach since we allow any origin. For more complete CORS handling
+        // a module like zfr-cors could be used.
+        $response = $this->getResponse();
+        $headers = $response->getHeaders();
+        $headers->addHeaderLine('Access-Control-Allow-Origin', '*');
+        $headers->addHeaderLine(
+            'Access-Control-Allow-Methods',
+            'GET, POST, OPTIONS'
+        );
+        $headers->addHeaderLine(
+            'Access-Control-Allow-Headers',
+            'Content-Type, Authorization, X-Requested-With'
+        );
+        $request = $this->getRequest();
+        if ($request->getMethod() == 'OPTIONS') {
+            return $this->output(null, 204);
+        }
+        return parent::onDispatch($e);
+    }
+
+    /**
+     * Save request data to auth_hash table.
+     *
+     * @return \Laminas\Http\Response
+     */
+    public function handshakeAction()
+    {
+        $requestParams = json_decode($this->getRequest()->getContent(), true);
+
+        $uuid = $this->getConfig()->Bazaar->uuid['moodle'];
```

Comment:
Jos asetusta ei ole määritelty, tämä pommittaa, kun uuid on null, eikö vain? Käsitellään jotenkin nätimmin esim. niin, että palauttaa vaikka `return $this->output([], self::STATUS_ERROR, 401, 'Unauthorized');`